### PR TITLE
Update dataprotocol client

### PR DESCRIPTION
**Pull request by anthonydresser**
removes the dataprotocol-node implementation we had and rewrites just the client part to use vscode's most up to date language client. Has some cleanup as well.

Functionally it is all the same expect I haven't added back the service waiting yet. I have another pull request that deals with this and handling cleaning up how we do that.

Needs Microsoft/sqltoolsservice#573

TODO:
- readd back in the service waiting
- move this to its own repo so others can use it
- make it so not every user of the client registers every feature

### CODE REVIEW

**Comment by kburtram** [comment]
@anthonydresser which version of the VS Code Language Protocol client did you base this one?  also, is what is the reason\issue for this change?  fixing something broken or just general code cleanup or something else?

File: extensions/mssql/client/src/controllers/mainController.ts [hunk 1]
```diff
@@ -33,6 +33,7 @@ export default class MainController implements vscode.Disposable {
 		vscodeWrapper?: VscodeWrapper) {
 		this._context = context;
 		this._vscodeWrapper = vscodeWrapper || new VscodeWrapper(MainController._extensionConstants);
+		this._vscodeWrapper.logToOutputChannel('logging');
```
Comment (kburtram) [inline extensions/mssql/client/src/controllers/mainController.ts]: should this be here?

**Comment by anthonydresser** [comment]
@kburtram This now just takes a npm dependency on vscode-languageclient, so it is in the package.json as version 3.5.0. Reason was to update the client code. They have add a few fixes/optimization as well as added the ability to dynamically register features, which could be useful in future implementation. This also removes a lot of src from our code base.

File: extensions/mssql/client/src/controllers/mainController.ts (hunk 1 above)
Comment (anthonydresser) [reply extensions/mssql/client/src/controllers/mainController.ts]: nope, will fix

File: extensions/mssql/client/src/typings/ref.d.ts [hunk 2]
```diff
@@ -4,5 +4,6 @@
  *--------------------------------------------------------------------------------------------*/
 
 /// <reference path='../../../../../src/vs/vscode.d.ts'/>
+/// <reference path='../../../../../src/vs/vscode.proposed.d.ts'/>
```
[the remark this answers is not in the document]
Comment (anthonydresser) [reply extensions/mssql/client/src/typings/ref.d.ts]: no, I don't remember why this is here

**Comment by anthonydresser** [comment]
@MattIrv the protocol hasn't changed so the server side still works with the updated client. The only change needed is I change the contract names for backup and restore to separate them from each other which already has a pr Microsoft/sqltoolsservice#573

**Pull request merged**